crates/client/rpc/src/lib.rs

```diff
@@ -215,6 +215,18 @@ where
         })
     }
 
+    /// Return the block hash of a given block number.
+    ///
+    /// # Arguments
+    ///
+    /// * `block_number` - The number of the block you'd like to retrieve the hash (starknet block).
+    fn get_cached_block_hash_by_number(&self, block_number: u64) -> Option<StarkHash> {
```

Comment:
Is it efficient in case of restart?

Reply:
This code has been deleted as it was no longer in use and causing clippy errors.